### ISSUE TITLE
Left Control Key doesn't work on External Keyboard

### ISSUE THREAD

**Issue by khamer** [kovidgoyal/kitty#2362]
I'm at a loss, but I'm only seeing this problem with kitty - my other terminals/applications work fine.

Here's the output from `kitty --debug-keyboard` from hitting left control + l, which doesn't work:
```
Press xkb_keycode: 0x25 clean_sym: Control_L composed_sym: Control_L mods: none glfw_key: 341 (LEFT CONTROL) xkb_key: 65507 (Control_L)
on_key_input: glfw key: 341 native_code: 0xffe3 action: PRESS mods: 0x0 text: '' state: 0 sent key to child
Press xkb_keycode: 0x2e clean_sym: l composed_sym: l mods: none glfw_key: 76 (L) xkb_key: 108 (l)
on_key_input: glfw key: 76 native_code: 0x6c action: PRESS mods: 0x0 text: '' state: 0 sent key to child
Release xkb_keycode: 0x2e clean_sym: l mods: none glfw_key: 76 (L) xkb_key: 108 (l)
on_key_input: glfw key: 76 native_code: 0x6c action: RELEASE mods: 0x0 text: '' state: 0 ignoring as keyboard mode does not allow release events
Release xkb_keycode: 0x25 clean_sym: Control_L mods: none glfw_key: 341 (LEFT CONTROL) xkb_key: 65507 (Control_L)
on_key_input: glfw key: 341 native_code: 0xffe3 action: RELEASE mods: 0x0 text: '' state: 0 ignoring as keyboard mode does not allow release events
```

and here's the the same thing for right control + l, which does work:
```
Press xkb_keycode: 0x25 clean_sym: Control_L composed_sym: Control_L mods: none glfw_key: 341 (LEFT CONTROL) xkb_key: 65507 (Control_L)
on_key_input: glfw key: 341 native_code: 0xffe3 action: PRESS mods: 0x0 text: '' state: 0 sent key to child
Press xkb_keycode: 0x2e clean_sym: l composed_sym: l mods: ctrl glfw_key: 76 (L) xkb_key: 108 (l)
on_key_input: glfw key: 76 native_code: 0x6c action: PRESS mods: 0x2 text: '' state: 0 sent key to child
Release xkb_keycode: 0x25 clean_sym: Control_L mods: ctrl glfw_key: 341 (LEFT CONTROL) xkb_key: 65507 (Control_L)
on_key_input: glfw key: 341 native_code: 0xffe3 action: RELEASE mods: 0x2 text: '' state: 0 ignoring as keyboard mode does not allow release events
Release xkb_keycode: 0x2e clean_sym: l mods: none glfw_key: 76 (L) xkb_key: 108 (l)
on_key_input: glfw key: 76 native_code: 0x6c action: RELEASE mods: 0x0 text: '' state: 0 ignoring as keyboard mode does not allow release events
```

The meaningful difference I see is the first one shows `mods: none` and the second doesn't. Why would kitty do this? Any insights?

I'm using Manjaro/i3wm/kitty 0.16.0

**Comment by khamer** [comment]
If it helps, <ctrl-c> and <ctrl-d> only work with the left control, but not the right.

**Comment by kovidgoyal** [comment]
Well I'm afraid I dont have a lot of interest in debugging issues using multiple keyboards simultaneously, but patches are welcome. The relevant code is in xkb_glfw.c Most likely this has something to do with per keyboard contexts.

**Comment by kovidgoyal** [comment]
This patch might help: https://github.com/kovidgoyal/kitty/commit/c7c14f184cffb0849a8e1fb3c98b3875b7342d7e

if not let me know.

**Comment by khamer** [comment]
@kovidgoyal I built kitty from source from 5f6aa1b (master) and this still doesn't work - shift does (it did before) but alt and control don't.

Also, so I don't seem totally crazy, the reason I even found this bug was because I was using [a keyboard similar to this](https://www.amazon.com/Koolertron-Programmable-Mechanical-Keyboard-Ergonomic/dp/B07PZT3Z25), which shows up to the system as two keyboards. It's definitely a less popular use case. I did try adding more logging to try to figure out what change needs to happen but didn't get too far - I'll try to dig a little more again.

Thanks anyway!
